test/clamm.test.ts

```diff
@@ -774,56 +798,105 @@ describe('math tests', () => {
-      const { nextSqrtPrice, amountIn, amountOut, feeAmount } = (
-        await clamm.contractInstance.methods.computeSwapStep(params)
-      ).returns
-      expect(nextSqrtPrice).toEqual(currentSqrtPrice)
-      expect(amountIn).toEqual(0n)
-      expect(amountOut).toEqual(0n)
-      expect(feeAmount).toEqual(1n)
+      const params = { args: { expectedAmountOut: minAmount, slippage: minFee } }
+      const result = (await clamm.contractInstance.methods.calculateMinAmountOut(params)).returns
+      expect(result).toEqual(0n)
     }
+    // min amount max fee
     {
-      const currentSqrtPrice = 10n ** 24n
-      const targetSqrtPrice = 1004987562112089027021926n
-      const liquidity = 2000n * 10n ** 5n
-      const amount = 20n
-      const fee = 60000n
-      {
-        const paramsIn = {
-          args: { currentSqrtPrice, targetSqrtPrice, liquidity, amount, byAmountIn: true, fee }
+      const params = { args: { expectedAmountOut: minAmount, slippage: maxFee } }
+      const result = (await clamm.contractInstance.methods.calculateMinAmountOut(params)).returns
+      expect(result).toEqual(0n)
+    }
+    // max amount max fee
+    {
+      const params = { args: { expectedAmountOut: maxAmount, slippage: maxFee } }
+      const result = (await clamm.contractInstance.methods.calculateMinAmountOut(params)).returns
+      expect(result).toEqual(0n)
+    }
+    // max amount min fee
+    {
+      const params = { args: { expectedAmountOut: maxAmount, slippage: minFee } }
+      const result = (await clamm.contractInstance.methods.calculateMinAmountOut(params)).returns
+      expect(result).toEqual(maxAmount)
+    }
+  })
+
+  test('is enough to change price - domain', async () => {
+    const uints = await deployUints(sender)
+    const clamm = await deployCLAMM(sender, uints.contractInstance.contractId)
+    const zeroLiquidity = 0n
+    const maxFee = 10n ** 12n
+    const maxAmount = (1n << 256n) - 1n
+    const minAmount = 1n
+    const minLiquidity = 1n
+    const maxSqrtPrice = 65535383934512647000000000000n
+    const minFee = 0n
+    // max fee
+    {
+      const params = {
+        args: {
+          amount: minAmount,
+          startingSqrtPrice: maxSqrtPrice,
+          liquidity: minLiquidity,
+          fee: maxFee,
+          byAmountIn: false,
+          xToY: false
         }
-        const { nextSqrtPrice, amountIn, amountOut, feeAmount } = (
-          await clamm.contractInstance.methods.computeSwapStep(paramsIn)
-        ).returns
-        expect(nextSqrtPrice).toEqual(targetSqrtPrice)
-        expect(amountIn).toEqual(10n)
-        expect(amountOut).toEqual(9n)
-        expect(feeAmount).toEqual(1n)
       }
-      {
-        const paramsOut = {
-          args: { currentSqrtPrice, targetSqrtPrice, liquidity, amount, byAmountIn: false, fee }
+      await expectError(clamm.contractInstance.methods.isEnoughToChangePrice(params))
```

Comment:
The name of this function is confused IMO.
I suggest to change `isEnoughToChangePrice` -> `isEnoughAmountToChangePrice`